# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/AudioConduit.cpp
# Commit: a802c8f458cb
# Full Hash: a802c8f458cb810df14eac3e7a1ab0c36124c277
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 1
# Description:
#   Bug 1654112 - Switch AudioConduit audio thread locking to MutexAutoTryLock. r=padenot
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D124367
# ==============================================================================

diff -r 0920243940ad -r a802c8f458cb dom/media/webrtc/libwebrtcglue/AudioConduit.cpp
--- a/dom/media/webrtc/libwebrtcglue/AudioConduit.cpp	Tue Aug 24 11:25:19 2021 +0200
+++ b/dom/media/webrtc/libwebrtcglue/AudioConduit.cpp	Wed Aug 25 16:44:09 2021 +0200
@@ -443,7 +443,8 @@
   }
 
   // If the Mutex is taken, skip this chunk to avoid blocking the audio thread.
-  if (!mMutex.TryLock()) {
+  MutexAutoTryLock tryLock(mMutex);
+  if (!tryLock) {
     CSFLogError(LOGTAG, "%s Conduit going through negotiation ", __FUNCTION__);
     return kMediaConduitPlayoutError;
   }
@@ -451,7 +452,6 @@
   // Conduit should have reception enabled before we ask for decoded
   // samples
   if (!mRecvStreamRunning) {
-    mMutex.Unlock();
     CSFLogError(LOGTAG, "%s Engine not Receiving ", __FUNCTION__);
     return kMediaConduitSessionNotInited;
   }
@@ -463,8 +463,6 @@
   auto info = static_cast<webrtc::internal::AudioReceiveStream*>(mRecvStream)
                   ->GetAudioFrameWithInfo(samplingFreqHz, frame);
 
-  mMutex.Unlock();
-
   if (info == webrtc::AudioMixer::Source::AudioFrameInfo::kError) {
     CSFLogError(LOGTAG, "%s Getting audio frame failed", __FUNCTION__);
     return kMediaConduitPlayoutError;
